### PR TITLE
Update put_s3_bucket_encryption.fp

### PR DESCRIPTION
**Pull request by gcasilva**
Fixing source file name for s3 bucket encryption function

### Checklist
- [x] Issue(s) linked - https://github.com/turbot/flowpipe-mod-aws/issues/69

### CODE REVIEW

**Comment by misraved** [comment]
Hi @gcasilva,

Thank you so much for the fix! 👍

I have a couple of questions and suggestions regarding the PR:

- To maintain consistency, I suggest we update the folder name under functions from `update_s3_bucket_encryption` to `put_s3_bucket_encryption`.
- Is there a particular reason you used `source = ".flowpipe/mods/github.com/turbot/flowpipe-mod-aws@v0.4.1/pipelines/s3/functions/update_s3_bucket_encryption"` instead of `./pipelines/s3/functions/put_s3_bucket_encryption`? I made this change locally to use `source = "./pipelines/s3/functions/put_s3_bucket_encryption"`, and it seemed to work fine.

Please feel free to reach out if you have any further questions. I'm happy to help! 👍

**Comment by gcasilva** [comment]
Hi @misraved please check my comments below:

1. Made the changes to the folder so its `put_s3_bucket_encryption `
2. `source = ".flowpipe/mods/github.com/turbot/flowpipe-mod-aws@v0.4.1/pipelines/s3/functions/update_s3_bucket_encryption"` is needed as when user runs `flowpipe mod install github.com/turbot/flowpipe-mod-aws` to install the flowpipe aws mod dependency it installs in this specific path convention `.flowpipe/mods/github.com/turbot/flowpipe-mod-aws@v0.4.1/` as tested by me and documented in https://flowpipe.io/docs/build/mod-dependencies
If we don't have this source path corrected user would have to manually create this directory in their current path or they would run into an error where the flowpipe function wouldn't find the source whenever they try to use the s3 bucket encryption fp, which is what happened to me before this fix.

The only caveat with this is that whenever flowpipe-mod-aws has a new release you'd have to change this line of code to reflect whatever new version is (e.g. v0.4.2) `.flowpipe/mods/github.com/turbot/flowpipe-mod-aws@v0.4.2/`

A better approach would be to change flowpipe logic to use the tag 'latest' for release instead of a version number which is what is uses today, but that's outside of what I can help on this PR.

**Comment by judell** [comment]
@gcasilva I'm not seeing that problem. I am in `~/flowpipe-mod-aws` with these changes only.

1. rename `functions/update_s3_bucket_encryption` -> `functions/put_s3_bucket_encryption`

2. 

```
-      ["--server-side-encryption-configuration", jsonencode(step.function.build_encryption_config.result)],
+      ["--server-side-encryption-configuration", jsonencode(step.function.build_encryption_config.response)],
```

Without these changes, this command fails as you reported.

```
flowpipe pipeline run aws.pipeline.put_s3_bucket_encryption   --arg 'region=us-east-1'   --arg 'bucket=jon-turbot-test-bucket-01'   --arg 'sse_algorithm=AWS256'   --arg 'bucket_key_enabled=true'`
```

With the changes, it succeeds.

What am I missing?

**Comment by judell** [comment]
OK I see the problem. 

If I run the command directly from the tweaked `~/flowpipe-mod-aws` it's OK.

But:

```
mkdir apn
cd apn 
flowpipe mod install ~/flowpipe-mod-aws
```

Now we are back to 

```
Bad Request: 'source' not found for function: aws.pipeline.put_s3_bucket_encryption.function.build_encryption_config
```

This indeed appears to be a Flowpipe bug that's beyond the scope of this PR.

@misraved, @khushboo9024 can we merge this PR without that change, and address other issue separately?

**Comment by johnsmyth** [comment]
@judell @misraved, @khushboo9024 @gcasilva 

FYI - I believe the bug with the relative path in `source` is fixed in 0.8.1 (thanks @vhadianto!): 
https://flowpipe.io/changelog/flowpipe-cli-v0-8-1

**Comment by judell** [comment]
Excellent, thanks @vhadianto!

File: pipelines/s3/put_s3_bucket_encryption.fp [hunk 1]
```diff
@@ -39,7 +40,7 @@ pipeline "put_s3_bucket_encryption" {
   }
 
   step "function" "build_encryption_config" {
-    source  = "./pipelines/s3/functions/put_s3_bucket_encryption"
+    source  = ".flowpipe/mods/github.com/turbot/flowpipe-mod-aws@v0.4.1/pipelines/s3/functions/put_s3_bucket_encryption"
```
Comment (misraved) [inline pipelines/s3/put_s3_bucket_encryption.fp:43]:
```suggestion
    source  = "./pipelines/s3/functions/put_s3_bucket_encryption"
```

**Pull request merged**